tests/installation/yast2_firstboot.pm

```diff
@@ -85,14 +86,14 @@ sub firstboot_registration {
 sub run {
     my $self      = shift;
     my $test_data = get_test_suite_data();
-    my %clients;
+    my $suffix    = $test_data->{custom_control_file} ? "_custom" : undef;
     foreach my $client (@{$test_data->{clients}}) {
         # Make sure the subroutine called from test data exists
         die "Client '$client' is not defined in the module, please check test_data" unless defined(&{"$client"});
         my $client_method = \&{"$client"};
-        $client_method->($self);
+        $client_method->($self, $suffix);
```

Comment:
I would recommend to use a `pre_run_hook` and there initialize a global variable to this file with a name more easy to read, like firstboot_type for example or is_custome_firstboot, and then use that variable in the methods. In that way you don't need to pass this arg to functions which do not use it.
Otherwise at least I would try to come up with other name than suffix.wdyt?

Reply:
Aren't global variable not recommended in general ?

Reply:
just trying to think how to avoid to pass parameter when we are calling the function dynamically to not make it more complex. Try to find the less evil here :) so maybe it is more simple in the way you have it. Take it as optional.

Reply:
I propose to change to $custom_needle, to make it more clear.